flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/operations/SqlDdlToOperationConverterTest.java

```diff
@@ -244,6 +262,233 @@ public void testAlterDatabase() throws Exception {
                 .isEqualTo(properties);
     }
 
+    @Test
+    public void testCreateModel() {
+        final String sql =
+                "CREATE MODEL model1 \n"
+                        + "INPUT(a bigint comment 'column a', b varchar, c int, d varchar)\n"
+                        + "OUTPUT(e bigint, f int)\n"
+                        + "  with (\n"
+                        + "    'task' = 'clustering',\n"
+                        + "    'provider' = 'openai',\n"
+                        + "    'openai.endpoint' = 'someendpoint'\n"
+                        + ")\n";
+        FlinkPlannerImpl planner = getPlannerBySqlDialect(SqlDialect.DEFAULT);
+        final CalciteParser parser = getParserBySqlDialect(SqlDialect.DEFAULT);
+        Operation operation = parse(sql, planner, parser);
+        assertThat(operation).isInstanceOf(CreateModelOperation.class);
+        CreateModelOperation op = (CreateModelOperation) operation;
+        CatalogModel catalogModel = op.getCatalogModel();
+        assertThat(catalogModel.getOptions())
+                .isEqualTo(
+                        ImmutableMap.of(
```

Comment:
```suggestion
                        Map.of(
```
nit: now Flink moved to jdk 11+ it's better to use jdk's `Map`/`List`